react/components/organisms/ripe-image-zoom-hover/ripe-image-zoom-hover.js

```diff
@@ -0,0 +1,154 @@
+import PropTypes from "prop-types";
+
+import RipeImageZoom from "../ripe-image-zoom/ripe-image-zoom";
+
+import "./ripe-image-zoom-hover.css";
+
+export class RipeImageZoomHover extends Component {
+    static get propTypes() {
+        return {
+            /**
+             * Zoom percentage that controls the level of zoom over the original image.
+             */
+            zoom: PropTypes.number,
+            /**
+             * The maximum zoom percentage allowed over the original image.
+             */
+            maxZoom: PropTypes.number,
+            /**
+             * The minimum zoom percentage allowed over the original image.
+             */
+            minZoom: PropTypes.number,
+            /**
+             * Enable changing the zoom value with the mouse wheel scroll.
+             */
+            scrollZoom: PropTypes.bool,
+            /**
+             * Scroll sensitivity when controlling the zoom value with the mouse wheel scroll.
+             */
+            scrollSensitivity: PropTypes.number,
+            /**
+             * Enables zooming out of the image with the mouse scroll.
+             */
+            zoomOut: PropTypes.bool
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            zoom: 140,
+            maxZoom: 300,
+            minZoom: 10,
+            scrollZoom: false,
+            scrollSensitivity: 1,
+            zoomOut: false
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            hover: false,
+            zoomData: this.props.zoom,
+            pivot: null
+        };
+    }
+
+    startHover(event, target) {
+        const pivotCoordinates = this._getPivotCoordinates(event, target);
+        this.setState({
+            hover: true,
+            pivot: pivotCoordinates,
+            zoomData: this.props.zoom
+        });
+    }
+
+    moveHover(event, target) {
+        if (!this.state.hover) return;
+
+        const pivotCoordinates = this._getPivotCoordinates(event, target);
+        this.setState({
+            pivot: pivotCoordinates
+        });
+    }
+
+    endHover() {
+        this.setState({
+            hover: false,
+            pivot: null,
+            zoomData: this.props.zoom
+        });
+    }
+
+    zoomScroll(event) {
+        // checks if zooming on hover is enabled or if the mouse is
+        // hovering the image
+        if (!this.state.hover || !this.props.scrollZoom) return;
+
+        const zoomValue = this.state.zoomData + -1 * this.props.scrollSensitivity * event.deltaY;
+
+        // checks if the zooming out feature is disabled, if so only
+        // allow zooming out until the base scaling of the image (100%)
+        if (!this.props.zoomOut && zoomValue <= 100) {
+            this.setState({
+                zoomData: 100
+            });
+            return;
+        }
+
+        // does not allow reducing to a zoom value smaller than 10 which

```

Comment:
Please fix the comment (hardcoded `10`). Also, I forgot to add `minZoom` to the docs so please do.

Reply:
Noticed now that there were no docs for the hover component, already creating them